src/lexer/lexer.py

```diff
@@ -78,11 +78,11 @@ def make_tokens(self) -> tuple[list[Token], None | Error]:
 
             elif self.current_char in DIGITS + '.':  # the char is a digit: we generate a number
                 there_is_a_space_or_a_tab_or_a_comment = False
-                number, error = self.make_number()
-                if error is None:  # there is no error
-                    tokens.append(number)
+                number_with_error = self.make_number()
```

Comment:
Why? I prefer `number, error = …`, it is more clear…

Reply:
The problem with `number, error` is that it breaks the link between the two. If you use a tuple, you can type hint it as `tuple[Token, None] | tuple[None, Error]`, which prevents the cases where there is a token and an error, and the case where there is none of them

Reply:
Is there really no other solution?